tests/sentry/explore/endpoints/test_explore_saved_queries.py

```diff
@@ -292,6 +292,115 @@ def test_get_starred_queries(self):
         assert len(response.data) == 1
         assert response.data[0]["name"] == "Starred query"
         assert response.data[0]["starred"] == 1
+        assert response.data[0]["position"] == 1
+
+    def test_get_most_starred_queries(self):
+        query = {"range": "24h", "query": [{"fields": ["span.op"], "mode": "samples"}]}
+        model = ExploreSavedQuery.objects.create(
+            organization=self.org,
+            created_by_id=self.user.id,
+            name="Most starred query",
+            query=query,
+        )
+        second_model = ExploreSavedQuery.objects.create(
+            organization=self.org,
+            created_by_id=self.user.id,
+            name="Second most starred query",
+            query=query,
+        )
+        model.set_projects(self.project_ids)
+        second_model.set_projects(self.project_ids)
+        ExploreSavedQueryStarred.objects.create(
+            organization=self.org,
+            user_id=self.user.id,
+            explore_saved_query=model,
+            position=1,
+        )
+        ExploreSavedQueryStarred.objects.create(
+            organization=self.org,
+            user_id=self.user.id + 1,
+            explore_saved_query=model,
+            position=1,
+        )
+        ExploreSavedQueryStarred.objects.create(
+            organization=self.org,
+            user_id=self.user.id,
+            explore_saved_query=second_model,
+            position=2,
+        )
+
+        with self.feature(self.feature_name):
+            response = self.client.get(self.url, data={"sortBy": "mostStarred"})
+        assert response.status_code == 200, response.content
+        assert len(response.data) == 3
+        assert response.data[0]["name"] == "Most starred query"
+        assert response.data[0]["starred"] == 1
```

Comment:
Can these be treated as both ints and booleans? I thought your `get_attrs` was setting these as bools. I would expect a boolean here instead

Reply:
updated to assert on booleans